web_external/js/views/body/UploadDatasetView.js

```diff
@@ -0,0 +1,276 @@
+                }
+                anonymous.prop('disabled', true);
+            }
+        },
+        'submit #isic-dataset-form': function (event) {
+            event.preventDefault();
+            this.$('#isic-dataset-submit').prop('disabled', true);
+            this.submitDataset();
+        }
+    },
+
+    initialize: function (settings) {
+        girder.cancelRestRequests('fetch');
+
+        this.uploadedZipFiles = [];
+        this.uploadedCsvFiles = [];
+        this.uploadFolderId = null;
+
+        this.render();
+    },
+
+    render: function () {
+        this.$el.html(isic.templates.uploadDataset({
+            user: girder.currentUser
+        }));
+
+        if (!this.uploadWidget) {
+            this.initializeUploadWidget();
+        }
+        this.updateUploadWidget();
+
+        this.$('input#isic-dataset-name').focus();
+
+        return this;
+    },
+
+    initializeUploadWidget: function () {
+        if (this.uploadWidget) {
+            this.stopListening(this.uploadWidget);
+            this.uploadWidget.destroy();
+        }
+        this.uploadWidget = new girder.views.UploadWidget({
+            parentView: this,
+            modal: false,
+            noParent: true,
+            title: false,
+            overrideStart: true
+        });
+
+        this.uploadWidget.setElement(this.$('.isic-upload-widget-container'));
+
+        this.listenTo(this.uploadWidget, 'g:filesChanged', this.filesSelected);
+        this.listenTo(this.uploadWidget, 'g:uploadStarted', this.uploadStarted);
+        this.listenTo(this.uploadWidget, 'g:uploadFinished', this.uploadFinished);
+    },
+
+    filesSelected: function (files) {
+        // TODO: could validate based on file extension
+    },
+    uploadStarted: function (files) {
+        // Prepare isic_upload folder in user's home and start upload
+        var uploadFolder = null;
+        var folderCollection = new girder.collections.FolderCollection();
+        folderCollection.once('g:changed', function () {
+            if (folderCollection.isEmpty()) {
+                // Create isic_upload folder
+                uploadFolder = new girder.models.FolderModel({
+                    name: 'isic_upload',
+                    parentType: 'user',
+                    parentId: girder.currentUser.get('_id'),
+                    description: 'ISIC uploads'
+                });
+
+                uploadFolder.once('g:saved', function () {
+                    this.startUpload(uploadFolder);
+                }, this).once('g:error', function () {
+                    girder.events.trigger('g:alert', {
+                        icon: 'cancel',
+                        text: 'Could not create upload folder.',
+                        type: 'error',
+                        timeout: 4000
+                    });
+                }, this).save();
+            } else {
+                // Use existing isic_upload folder, but clean it first
```

Comment:
This cleanup logic is probably unnecessary right now, but feel free to leave it, as I'm sure it will eventually be used.